src/@batch-flask/ui/virtual-scroll/virtual-scroll.component.ts

```diff
@@ -358,7 +358,8 @@ export class VirtualScrollComponent implements OnInit, AfterViewInit, OnChanges,
     }
 
     private _detectViewportChanges(items: any, start: number, end: number, forceViewportUpdate: boolean) {
-        if (start !== this.previousStart || end !== this.previousEnd || forceViewportUpdate === true) {
+        const minNum = this._calculateDimensions().itemsPerCol;
```

Comment:
Not sure that the right fix for this. Does this not affect performance if you are scrolling in the page. How often does this method gets called. Need to be quite carful about this component as performance is its main point and you want to limit computation that happens too often.

Also I would not call calculateDim here but just pass itemsPerCol as its only called form a place which already just calulated those dimenstions.

Reply:
Its called a fair bit, but didn't see any large performance impact through manual testing. I see the point though.  I'll abandon this PR, looks like we should be able to just change the fetch/fetchNext processItems method to recursively call fetchNext if there are no items, which should be less intensive.